cmd/launcher/secure_enclave_dawin.go

```diff
@@ -0,0 +1,161 @@
+		if err != nil {
+			return fmt.Errorf("parsing test server public key: %w", err)
+		}
+
+		serverPubKeys[string(secureenclavesigner.TestServerPubKey)] = k
+	}
+
+	for _, keyStr := range []string{certs.K2EccServerCert, certs.ReviewEccServerCert, certs.LocalhostEccServerCert} {
+		key, err := echelper.PublicPemToEcdsaKey([]byte(keyStr))
+		if err != nil {
+			return fmt.Errorf("parsing server public key from pem: %w", err)
+		}
+
+		pubB64Der, err := echelper.PublicEcdsaToB64Der(key)
+		if err != nil {
+			return fmt.Errorf("marshalling server public key to b64 der: %w", err)
+		}
+
+		serverPubKeys[string(pubB64Der)] = key
+	}
+
+	switch args[0] {
+	case "create-key":
+		return createSecureEnclaveKey(args[1])
+
+	case "sign":
+		return signWithSecureEnclave(args[1])
+
+	default:
+		return fmt.Errorf("unknown command %s", args[0])
+	}
+}
+
+func createSecureEnclaveKey(requestB64 string) error {
+	b, err := base64.StdEncoding.DecodeString(requestB64)
+	if err != nil {
+		return fmt.Errorf("decoding b64 request: %w", err)
+	}
+
+	var request secureenclavesigner.Request
+	if err := msgpack.Unmarshal(b, &request); err != nil {
+		return fmt.Errorf("unmarshaling msgpack request: %w", err)
+	}
+
+	if err := verifySecureEnclaveChallenge(request); err != nil {
+		return fmt.Errorf("verifying challenge: %w", err)
+	}
+
+	secureEnclavePubKey, err := secureenclave.CreateKey()
+	if err != nil {
+		return fmt.Errorf("creating secure enclave key: %w", err)
+	}
+
+	secureEnclavePubDer, err := echelper.PublicEcdsaToB64Der(secureEnclavePubKey)
+	if err != nil {
+		return fmt.Errorf("marshalling public key to der: %w", err)
+	}
+
+	fmt.Println(string(secureEnclavePubDer))
+	return nil
+}
+
+func signWithSecureEnclave(signRequestB64 string) error {
+	b, err := base64.StdEncoding.DecodeString(signRequestB64)
+	if err != nil {
+		return fmt.Errorf("decoding b64 sign request: %w", err)
+	}
+
+	var signRequest secureenclavesigner.SignRequest
+	if err := msgpack.Unmarshal(b, &signRequest); err != nil {
+		return fmt.Errorf("unmarshaling msgpack sign request: %w", err)
+	}
+
+	if err := verifySecureEnclaveChallenge(signRequest.Request); err != nil {
+		return fmt.Errorf("verifying challenge: %w", err)
+	}
+
+	secureEnclavePubKey, err := echelper.PublicB64DerToEcdsaKey(signRequest.SecureEnclavePubKey)
+	if err != nil {
+		return fmt.Errorf("marshalling b64 der to public key: %w", err)
+	}
+
+	ses, err := secureenclave.New(secureEnclavePubKey)
+	if err != nil {
+		return fmt.Errorf("creating secure enclave signer: %w", err)
+	}
+	var sig []byte
+	backoff.WaitFor(func() error {
+		sig, err = ses.Sign(rand.Reader, signRequest.Digest, crypto.SHA256)
+		return err
+	}, 250*time.Millisecond, 2*time.Second)
+
+	if err != nil {
+		return fmt.Errorf("signing request: %w", err)
+	}
+
+	fmt.Print(base64.StdEncoding.EncodeToString(sig))
+	return nil
+}
+
+func verifySecureEnclaveChallenge(request secureenclavesigner.Request) error {
```

Comment:
Would we want to also find and check our parent process to make sure it's launcher?

Reply:
hmm, maybe, were validating that we have a recently signed request from the server maybe that's enough? I'm also not sure how we could be confident that it's actually the launcher binary unless we validate the codesign signature, which would make testing more complicated. I'm open to discussion on the topic.

Reply:
It's probably enough, and maybe this would be overkill -- but maybe worth it to be very cautious?

If we think it might be worth the caution and complicating testing is the only drawback, I would be super happy to get the automated tests going for macOS so we can see how much easier it'd be there with an already-signed app bundle (i.e. merging this PR as-is and then revisiting the discussion after, once we know more).

Reply:
I believe we settled on just having the k2 signed request, re open if disagree